x-pack/legacy/plugins/siem/public/pages/detection_engine/rules/create/index.tsx

```diff
@@ -117,8 +116,7 @@ const CreateRulePageComponent: React.FC = () => {
       getActionMessageParams((stepsData.current['define-rule'].data as DefineStepRule).ruleType),
     [stepsData.current['define-rule'].data]
   );
-  const userHasNoPermissions =
-    canUserCRUD != null && hasManageApiKey != null ? !canUserCRUD || !hasManageApiKey : false;
+  const userHasNoPermissions = canUserCRUD != null ? !canUserCRUD : false;
```

Comment:
This small chunk of code should have test coverage and this should be potentially a pure function.

I see this chunk being copied in multiple places which is bad. I would pull this out into a pure function in a utils/helper, add tests and then call it here and below.

Reply:
what about just using `!canUserCRUD` and you already have test for that!